server/controllers/categorised/employees.js

```
@@ -63,16 +99,125 @@ exports.checkHoliday = function (req, res, next) {
+  'fonction.id AS fonction_id, fonction.fonction_txt, service.name AS service_txt, ' +
+  'employee.phone, employee.email, employee.adresse, employee.bank, employee.bank_account, employee.daily_salary, employee.location_id, ' +
+  'grade.code AS code_grade, debitor.uuid as debitor_uuid, debitor.text AS debitor_text,debitor.group_uuid as debitor_group_uuid, ' +
+  'creditor.uuid as creditor_uuid, creditor.text AS creditor_text, creditor.group_uuid as creditor_group_uuid, creditor_group.account_id ' +
+  'FROM employee ' +
+  ' JOIN grade ON employee.grade_id = grade.uuid ' +
+  ' JOIN fonction ON employee.fonction_id = fonction.id ' +
+  ' JOIN debitor ON employee.debitor_uuid = debitor.uuid ' +
+  ' JOIN creditor ON employee.creditor_uuid = creditor.uuid ' +
+  ' JOIN creditor_group ON creditor_group.uuid = creditor.group_uuid ' +
+  ' LEFT JOIN service ON service.id = employee.service_id ' +
+  'WHERE employee.id = ? ';
+
+  db.exec(sql, [req.params.id])
+  .then(function (rows) {
+    // send a 204 if rows is empty
+    if (!rows.length) {
+      return res.status(204).json({
+        code   : 'SUCCES_NO_CONTENT',
+        reason : 'No employee found by id ' + req.params.id
+      });
+    }
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+};
+
+/**
+* Update details of an employee referenced by an `id` in the database
+*
+* @param {object} request The express request object
+* @param {object} response The express response object
+* @param {object} next The express middleware next object
+*
+* @example
+* // PUT /employees/:id : Update details of an employee
+* var employees = require('categorised/employees');
+* employees.update(request, response, next);
+*/
+exports.update = function (req, res, next) {
+  var sql = 'UPDATE employee SET ? WHERE employee.id = ?';
+
+  db.exec(sql, [req.body, req.params.id])
+  .then(function (row) {
+    if (!row.affectedRows) { return res.status(204).json(row); }
```

This should return an error code.